packages/plugin-core/src/commands/ShowPreview.ts

```diff
@@ -216,6 +217,68 @@ const handleAssetLink = async ({
   await ShowPreviewAssetOpener.openWithDefaultApp(assetFullPath);
 };
 
+export const getNavigationTargetNoteForWikiLink = async ({
+  data,
+  notes,
+}: {
+  data: NoteViewMessage["data"];
+  notes: NotePropsDict;
+}) => {
+  // wiki links will have the following format
+  //
+  // with `prettyLinks` set to false
+  //    with anchor: vscode-webview://4e98b9cf-41d8-49eb-b458-fcfda32c6c01/foo.html'
+  //
+  // with `prettyLinks` set to true
+  //    with anchor: vscode-webview://4e98b9cf-41d8-49eb-b458-fcfda32c6c01/foo'
+  //    without anchor: vscode-webview://4e98b9cf-41d8-49eb-b458-fcfda32c6c01/foo#foobar'
+  //
+  // And when the target of the link is a note in different vault without specifying
+  // the vault explicitly the href is going to have the file name of the node in place of the id:
+  // Example of href note inf different vault without vault specified:
+  // vscode-webview://25d7783e-df29-479c-9838-386c17dbf9b6/dendron.ref.links.target-different-vault
+  //
+  if (!data.href) {
+    throw ErrorFactory.createInvalidStateError({
+      message: `href is missing from data: '${ErrorFactory.safeStringify(
+        data
+      )}'`,
+    });
+  }
+
+  const noteId = extractNoteIdFromHref(data);
+  if (!noteId) {
+    throw ErrorFactory.createInvalidStateError({
+      message: `Failed to extract noteId from '${ErrorFactory.safeStringify(
+        data
+      )}'`,
+    });
+  }
+
+  const anchor = extractHeaderAnchorIfExists(data.href);
+  let note: NoteProps | undefined = notes[noteId];
+
+  if (note === undefined) {
+    // If we could not find the note by the extracted id there is a good chance that the name
+    // of the note was in place of the id in the HREF (as in case of navigating to a note
+    // in a different vault without explicit vault specification). Hence we will attempt
+    // to find the note by file name.
+    const candidates = NoteUtils.getNotesByFname({ fname: noteId, notes });
+
+    // In case of multiple candidates matching the note we will grab the one with oldest (smallest)
+    // create date to make the navigation decently predictable (the predictability
+    // can be thrown off if a new vault is added to the workspace with matching name and older note).
+    if (candidates.length > 0) {
```

Comment:
Do you think it's worth warning the user if we have more than 1 candidate? Like an info box saying something along the links of "Ambiguous Link - ${noteCount} notes exist with this name across ${vaultCount} vaults. Defaulting to oldest note."

Reply:
The downside of the info box, that it doesn't go away on its own. But yea could be worthwhile to add. 

Was also considering showing a quick pick with the vaults however, that seemed a bit odd user experience to attempt to prop open a quick pick of vaults after a click.

Reply:
An alternative (and perhaps better) way we could handle this is to have a code warning whenever there's an ambiguous link; maybe even with a context UI for a quick right click > fix (ask user to specify the vault, and then we format the link for them).

Reply:
Yep the auto fix makes sense for subsequent versions.

For now added the info message when we are navigating to a link with duplicates in a different vault.

Did not add info message when navigating to a link in current vault (that has duplicates in other vault) since we cannot distinguish from the data that comes to us whether the vault was explicitly specified in the link or whether it is in the current vault link.